src/rust/grapl-web-sec/src/authn/dynamodb.rs

```diff
@@ -0,0 +1,237 @@
+use std::ops::Add;
+
+use argon2::{
+    PasswordVerifier,
+    Version,
+};
+use chrono::{
+    Duration as ChronoDuration,
+    TimeZone,
+    Utc,
+};
+use hmap::hmap;
+use rand::Rng;
+use rusoto_dynamodb::{
+    AttributeValue,
+    DynamoDb,
+    DynamoDbClient,
+    GetItemInput,
+    PutItemInput,
+};
+use serde::{
+    Deserialize,
+    Serialize,
+};
+use tap::tap::*;
+use tracing::{
+    error,
+    warn,
+};
+
+use crate::error::WebSecError;
+
+const EXPIRATION_TIMEOUT_DAYS: i64 = 1;
+const SESSION_TOKEN_LENGTH: usize = 32;
+
+pub(crate) async fn get_valid_session_row(
+    dynamodb_client: &DynamoDbClient,
+    session_token: String,
+) -> Result<SessionRow, WebSecError> {
+    let session_instance = hmap! {
+        "session_token".to_owned() => AttributeValue {
+            s: Some(session_token),
+            ..Default::default()
+        }
+    };
+
+    let session_query = GetItemInput {
+        consistent_read: Some(true),
+        key: session_instance,
+        table_name: grapl_config::user_session_table_name(),
+        ..Default::default()
+    };
+
+    let session_row_hashmap = dynamodb_client
+        .get_item(session_query)
+        .await?
+        .item
+        .ok_or(WebSecError::MissingSession)
+        .tap_err(|_| warn!("No entries for session found in session table."))?;
```

Comment:
1. It's been asked that anything we send to a tracing log macro go into a field, so if anything this should be something like `warn!(message="...")`, however:
2. In this particular case the error text should be attached to the error itself with the `#[error("...")]` enum variant macro from `thiserror`, which implements `Display` for it. So at this point, this should look more like:
```
tap_err(|error| warn!(message=%error))?;
```

Maybe use "error" for field name instead of "message", idk.